### PR TITLE
Jk/cumulus 3320 upgrade docs

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-3220](https://bugs.earthdata.nasa.gov/browse/CUMULUS-3220)

## Changes

* Adds upgrade documentation for manual database upgrade

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: docs/upgrade-notes/upgrade-execution-table-CUMULUS-3320.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+ Please note that testing in a production-similar environment is *strongly* advised if you are concerned about precise query times and/or downtime.
+
+The following procedures detail how this upgrade may be performed:
+
+## Adding the `executions_parent_cumulus_id_index` index
+
+### Reasoning
+
+This index is required to support effective deletion of records given query analysis implies the same table foreign-key constraint results in table scan to enforce the constraint for each deletion.
```
Comment (npauzenga) [inline docs/upgrade-notes/upgrade-execution-table-CUMULUS-3320.md:21]:
```suggestion
This index is required to support effective deletion of records. Our query analysis implies the same table foreign-key constraint results in a slow table scan to enforce the constraint for each deletion.
```

File: docs/upgrade-notes/upgrade-execution-table-CUMULUS-3320.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+To do this, use the following query:
+
+```sql
+CREATE INDEX executions_parent_cumulus_id_index ON executions(parent_cumulus_id)'
+```
+
+##### Create index concurrently/with ongoing:
+
+The required index can also be created while the database is in use prior to installing the upgrade containing CUMULUS-3320 by running the following query:
```
Comment (npauzenga) [inline docs/upgrade-notes/upgrade-execution-table-CUMULUS-3320.md:51]:
```suggestion
The required index can also be created while the database is in use *prior to installing the upgrade containing CUMULUS-3320* by running the following query:
```

File: docs/upgrade-notes/upgrade-execution-table-CUMULUS-3320.md [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+For reference, in testing, using a 4ACU Aurora Serverless V1 cluster on a table with 15 million rows this migration took roughly 3 minutes to complete.
+
+To do this, use the following query:
+
+```sql
+CREATE INDEX executions_collection_cumulus_id_index ON executions(collection_cumulus_id)
+```
+
+The required index can also be created while the database is in use prior to installing the upgrade containing CUMULUS-3320 by running the following query:
```
Comment (npauzenga) [inline docs/upgrade-notes/upgrade-execution-table-CUMULUS-3320.md:101]:
```suggestion
The required index can also be created while the database is in use *prior to installing the upgrade containing CUMULUS-3320* by running the following query:
```

File: docs/upgrade-notes/upgrade-execution-table-CUMULUS-3320.md [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+```sql
+BEGIN;
+ALTER TABLE executions DROP CONSTRAINT IF EXISTS executions_parent_cumulus_id_foreign;
+ALTER TABLE executions ADD CONSTRAINT executions_parent_cumulus_id_foreign FOREIGN KEY (parent_cumulus_id) REFERENCES executions(cumulus_id) ON DELETE SET NULL NOT VALID;
+END;
+```
+
+#### Re-validate the constraint
+
+This query can be run at any time following the above step and will not lock the full table for new row inserts/selects:
```
Comment (npauzenga) [inline docs/upgrade-notes/upgrade-execution-table-CUMULUS-3320.md:160]: Maybe a note about why we need to validate? Or is that too close to 'how to SQL'?

Comment (npauzenga) [reply docs/upgrade-notes/upgrade-execution-table-CUMULUS-3320.md:160]: Eh, it's mostly explained above.

**Review by npauzenga — CHANGES_REQUESTED** [review]
This looks really good @Jkovarik. I added one nit. 

This could be something we update once questions come up. It's clear to me having the context around these changes.

File: docs/upgrade-notes/upgrade-execution-table-CUMULUS-3320.md (hunk 4 above)
Comment (Jkovarik) [reply docs/upgrade-notes/upgrade-execution-table-CUMULUS-3320.md:160]: There's definitely an audience background concern here.   I think I fall on the side of YAGNI on this, but if you feel strongly I can try to include some links to postgres documents.

**Comment by Jkovarik** [comment]
@npauzenga  thanks for the updates 🙇🏻  -- I took the changes and responded to your comments, let me know what you think!

**Review by npauzenga — APPROVED** [review]
Looks great. Thanks @Jkovarik !

**Review by npauzenga — APPROVED** [review]
👍

**Pull request merged**